src/server/translate_cache.ts

```diff
@@ -128,6 +134,47 @@ export class TranslateCache implements TranslateCache {
     });
   }
 
+  async translateWithTruncatedCache(
+    document: TextDocument,
+    text: string,
+    exploreCount: number,
+    queriedSource: string
+  ): Promise<Model | undefined> {
+    const {uri, languageId} = document;
+    if (languageId === 'malloy') {
+      const entry = this.truncatedCache.get(uri);
+      if (entry && entry.exploreCount === exploreCount) {
+        return entry.model;
+      }
+      const files = {
+        readURL: (url: URL) => {
+          if (url.toString() === uri) {
+            return Promise.resolve(text);
+          } else {
+            return this.getDocumentText(this.documents, url);
+          }
+        },
+      };
+      // TODO: Look into having remaining statements run "in the background" and having new runs
+      //       preempt the current fetch
+      const runtime = new Runtime(
+        files,
+        this.connectionManager.getConnectionLookup(new URL(uri))
+      );
+      const mm = await this.createModelMaterializer(uri, runtime);
+      const model = await mm?.getModel();
```

Comment:
Sorry, I changed the return signature of this method, you'll want `const {modelMaterializer: mm} = `

Reply:
No worries and thanks for the heads up, resolved